### PR TITLE
fix NPE, issue 3335

### PR DESCRIPTION
**Pull request by pinges**
https://github.com/hyperledger/besu/issues/3335
Signed-off-by: Stefan Pingel <stefan.pingel@consensys.net>

### CODE REVIEW

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java [hunk 1]
```diff
@@ -279,6 +279,7 @@ protected boolean validatePacketSize(final int packetSize) {
   }
 
   protected void handleIncomingPacket(final Endpoint sourceEndpoint, final Packet packet) {
+    // TODO: check whether
```
Comment (macfarla) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java:282]: stray TODO

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/VertxPeerDiscoveryAgent.java [hunk 2]
```diff
@@ -148,26 +148,31 @@ protected void handleListenerSetup(
...
-          if (ar.failed()) {
-            result.completeExceptionally(ar.cause());
-          } else {
-            result.complete(null);
-          }
-        });
+    final CompletableFuture<Void> result = new CompletableFuture<>();
+    if (socket == null) {
+      result.completeExceptionally(
+          new RuntimeException("Discovery socket already closed, because Besu is closing down"));
```
Comment (frankisawesome) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/VertxPeerDiscoveryAgent.java:162]: Is a RuntimeException appropriate here or something more specific?

**Review by frankisawesome — CHANGES_REQUESTED** [review]
LGTM, but I would still like to see a test for this method here

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/VertxPeerDiscoveryAgent.java (hunk 2 above)
Comment (pinges) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/VertxPeerDiscoveryAgent.java:162]: This should only happen when we are closing down (we have called stop in the VertxPeerDiscoveryAgent, which sets the socket to null), but there is still a message in the system that causes  sendOutgoingMessage() to be called.

**Comment by mark-terry** [comment]
Does this specifically address the reported issue, or just add handling around the case where the socket may be null?

**Review by frankisawesome — APPROVED** [review]
Discussed on zoom, we will make an issue for missing test file for this class :)

**Pull request merged**